src/simulation/LoadGenerator.cpp

```diff
@@ -412,6 +424,26 @@ LoadGenerator::start(GeneratedLoadConfig& cfg)
     mPreLoadgenApplySorobanFailure =
         mTxGenerator.getApplySorobanFailure().count();
 
+    if (cfg.mode == LoadGenMode::PAY_PREGENERATED)
```

Comment:
Do we need to assert/fail if `GENESIS_TEST_ACCOUNT_COUNT` is 0 here? Similarly, should we fail is `GENESIS_TEST_ACCOUNT_COUNT` is set but someone attempts to invoke create accounts load?

Reply:
Yeah, extra checks sound good, added. In the change I left the door open for flexible uses like old-style account creation + pre-generated txs, or new-style account creation + old-style account creation with an offset. But those are weird/niche use cases I don't think we need to support anyway.